src/Sentry/ServerSessionManager.cs

```diff
@@ -0,0 +1,151 @@
+﻿using System;
+using System.Timers;
+using Sentry.Extensibility;
+using Sentry.Infrastructure;
+using Sentry.Internal;
+
+namespace Sentry
+{
+    // Strategy for future reference:
+    // - Keep only one set of sessions in memory
+    // - Make the timer run on-the-minute
+    // - Just track the exited/error count
+    // - When the timer hits - flush
+
+    // AKA server mode
+    // https://develop.sentry.dev/sdk/sessions
+    internal class ServerSessionManager : ISessionManager, IDisposable
+    {
+        private readonly object _lock = new();
+
+        private readonly SentryOptions _options;
+        private readonly ISentryClient _client;
+        private readonly ISystemClock _clock;
+        private readonly Timer _timer;
+
+        private int _exitedCount;
+        private int _erroredCount;
+
+        public ServerSessionManager(
+            SentryOptions options,
+            ISentryClient client,
+            ISystemClock clock)
+        {
+            _options = options;
+            _client = client;
+            _clock = clock;
+
+            // TODO: timer should be synced to run *on the minute*
+            _timer = new Timer {Interval = 60 * 1000, Enabled = true, AutoReset = true};
```

Comment:
This constructor is only compatible with .NET 5

Reply:
oh, Timer is implemented at two different places: System.Threading and System.Timers

Reply:
Actually more like 4 different places :joy: